core/azure-core/src/main/java/com/azure/core/implementation/configuration/Configuration.java

```diff
@@ -0,0 +1,262 @@
+    }
+
+    /**
+     * @return a clone of the Configuration object.
+     */
+    @SuppressWarnings("CloneDoesntCallSuperClone")
+    public Configuration clone() {
+        return new Configuration(configurations);
+    }
+
+    /**
+     * Attempts to convert the configuration value to {@code T}.
+     *
+     * If the value is null or empty then the default value is returned.
+     *
+     * @param value Configuration value retrieved from the map.
+     * @param defaultValue Default value to return if the configuration value is null or empty.
+     * @param <T> Generic type that the value is converted to if not null or empty.
+     * @return The converted configuration, if null or empty the default value.
+     */
+    @SuppressWarnings("unchecked")
+    private <T> T convertOrDefault(String value, T defaultValue) {
+        // Value is null or empty, return the default.
+        if (ImplUtils.isNullOrEmpty(value)) {
+            return defaultValue;
+        }
+
+        // Check the default value's type to determine how it needs to be converted.
+        Object convertedValue;
+        if (defaultValue instanceof Byte) {
+            convertedValue = Byte.parseByte(value);
+        } else if (defaultValue instanceof Short) {
+            convertedValue = Short.parseShort(value);
+        } else if (defaultValue instanceof Integer) {
+            convertedValue = Integer.parseInt(value);
+        } else if (defaultValue instanceof Long) {
+            convertedValue = Long.parseLong(value);
+        } else if (defaultValue instanceof Float) {
+            convertedValue = Float.parseFloat(value);
+        } else if (defaultValue instanceof Double) {
+            convertedValue = Double.parseDouble(value);
+        } else if (defaultValue instanceof Boolean) {
+            convertedValue = Boolean.parseBoolean(value);
+        } else {
+            convertedValue = value;
+        }
+
+        return (T) convertedValue;
+    }
+
+    /**
+     * Attempts to load the configuration using the passed loader. If the configuration is found it will be added to
+     * the configuration store and a message will be logged.
+     *
+     * @param name Name of the configuration.
+     * @param loader Loading function to apply.
+     * @param logMessage Message to log if the configuration is found.
+     * @return True if the configuration was loaded, false otherwise.
+     */
+    private boolean loadFrom(String name, Function<String, String> loader, String logMessage) {
+        String value = loader.apply(name);
+        if (!ImplUtils.isNullOrEmpty(value)) {
+            configurations.put(name, value);
+            logger.asInformational().log(logMessage, name);
+            return true;
+        }
+
+        return false;
+    }
+
+    /*
+     * Noop Configuration used to opt out of using global configurations when constructing client libraries.
+     */
+    private static class NoopConfiguration extends Configuration {
```

Comment:
You might want to override `clone` here to return a new instance or throw some useful exception.

Reply:
Returning a new instance